### PR TITLE
Eventarc update

### PR DESCRIPTION
**Pull request by slevenick**
Adds support for specifying a pubsub topic in the transport field. Not mergeable yet. Requires DCL fixes and downstream DCL updates

Fixes: https://github.com/hashicorp/terraform-provider-google/issues/8711

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-downstream-tools), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/master/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/master/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/master/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
eventarc: added `transport.pubsub.topic` to `google_eventarc_trigger` (beta only)
```
```release-note:new-resource
google_eventarc_trigger (GA only)
```

### CODE REVIEW

File: tpgtools/handwritten/utils.go [hunk 1]
```diff
@@ -1,10 +1,13 @@
+// Contains functions that don't really belong anywhere else.
```
Comment (rileykarson) [inline tpgtools/handwritten/utils.go:1]: Am I missing something, or is this file now managed by both mmv1+tpgtools? Should we delete the MMv1 one if so?

Comment (slevenick) [reply tpgtools/handwritten/utils.go:1]: Yes, good call! It gets overwritten anyways right now

File: tpgtools/overrides/eventarc/trigger.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+- type: CUSTOM_NAME
+  field: destination.cloud_run
+  details:
+    name: cloud_run_service
+- type: CUSTOM_NAME
+  field: event_filters
+  details:
+    name: matching_criteria
```
[the remark this answers is not in the document]
Comment (slevenick) [reply tpgtools/overrides/eventarc/trigger.yaml:8]: Yeah, API-side field renames. Starting discussion with them now

[the remark this answers is not in the document]
Comment (rileykarson) [reply tpgtools/overrides/eventarc/trigger.yaml:8]: Great, thanks! Since this is non-breaking in TF due to these overrides, non-blocking for now.

**Review by rileykarson — APPROVED** [review]

**Pull request merged**